### PR TITLE
IBP-1971 Exclude Null Parent Gid

### PR DESCRIPTION
**Pull request by gelli-leafnode**
Exclude null parent id by checking if the option is present 
Catch null pointer exception to list with null parent

### CODE REVIEW

File: src/test/java/org/generationcp/middleware/service/pedigree/PedigreeServiceImplTest.java [hunk 1]
```diff
@@ -106,7 +107,12 @@ public void testGetCrossExpansions() {
         final Germplasm generateRandomGermplasm = this.generateRandomGermplasmRecurringMaleParent();
         Set<Integer> set = new HashSet<>();
         set.add(generateRandomGermplasm.getGid());
-        Map<Integer, String> map = this.pedigreeService.getCrossExpansions(set, 5, this.crossExpansionProperties);
-        Assert.assertTrue(map.containsValue("-"));
+        try{
+            this.pedigreeService.getCrossExpansions(set, 5, this.crossExpansionProperties);
```
Comment (darla-leafnode) [inline src/test/java/org/generationcp/middleware/service/pedigree/PedigreeServiceImplTest.java]: Put Assert.fail after this line:
`this.pedigreeService.getCrossExpansions(set, 5, this.crossExpansionProperties);`

File: src/test/java/org/generationcp/middleware/service/pedigree/AncestryTreeServiceTest.java [hunk 2]
```diff
@@ -149,11 +149,57 @@ private Germplasm generateTestGermplasm(final int gid, final int methodId) {
 		germplasm.setGid(gid);
 		germplasm.setGnpgs(2);
 		final Method method = new Method(methodId);
+		if(methodId > 1) {
+			method.setMname("Backcross");
+			germplasm.setGpid1(0);
```
Comment (darla-leafnode) [inline src/test/java/org/generationcp/middleware/service/pedigree/AncestryTreeServiceTest.java]: Maybe better to use non-zero as parent GIDs?

**Comment by mderamos** [comment]
@gelli-leafnode  please resolve conflicts and check if the changes still works.

File: src/test/java/org/generationcp/middleware/service/pedigree/AncestryTreeServiceTest.java (hunk 2 above)
Comment (gelli-leafnode) [reply src/test/java/org/generationcp/middleware/service/pedigree/AncestryTreeServiceTest.java]: Removed this unit test. Unit test are in PedigreeServiceImplTest

**Pull request merged**